src/panels/lovelace/cards/hui-weather-forecast-card.ts

```diff
@@ -166,13 +166,22 @@ class HuiWeatherForecastCard extends LitElement implements LovelaceCard {
         : undefined;
 
     let hourly: boolean | undefined;
+    let multiday: boolean | undefined;
 
     if (forecast?.length && forecast?.length > 2) {
       const date1 = new Date(forecast[1].datetime);
       const date2 = new Date(forecast[2].datetime);
       const timeDiff = date2.getTime() - date1.getTime();
 
       hourly = timeDiff < DAY_IN_MILLISECONDS;
+
+      if (hourly) {
+        const date0 = new Date(forecast[0].datetime);
+        const datelast = new Date(forecast[forecast.length - 1].datetime);
+        const dayDiff = datelast.getTime() - date0.getTime();
+
+        multiday = dayDiff > DAY_IN_MILLISECONDS;
```

Comment:
A forecast array can be long, so is you would have more than 24 hourly forecasts this would also be true.

Maybe better to just take the 5 results we are showing?

Reply:
That makes sense. I will update just to use the five days for now. This whole logic can be replaced if the architecture issue is resolved, which I agree is a better approach. I didn't realize that change was being considered.